pypeit/utils.py

```diff
@@ -1361,15 +1373,20 @@ def robust_optimize(ydata, fitfunc, arg_dict, maxiter=10, inmask=None, sigma=Non
         iter += 1
 
     if (iter == maxiter) & (maxiter != 0):
-        msgs.warn('Maximum number of iterations maxiter={:}'.format(maxiter) + ' reached in sens_tell_fit')
+        msgs.warn('Maximum number of iterations maxiter={:}'.format(maxiter) + ' reached in robust_optimize')
     outmask = np.copy(thismask)
     if np.sum(outmask) == 0:
         msgs.warn('All points were rejected!!! The fits will be zero everywhere.')
 
-    # Perform a final fit using the final outmask
-    result, ymodel = fitfunc(ydata, outmask, arg_dict, **kwargs_optimizer)
+    # Perform a final fit using the final outmask if new pixels were rejected on the last iteration
+    if qdone is False:
+        ret_tuple = fitfunc(ydata, outmask, arg_dict, **kwargs_optimizer)
+    if (len(ret_tuple) == 2):
+        return ret_tuple[0], ret_tuple[1], outmask
+    elif (len(ret_tuple) == 3):
+        return ret_tuple[0], ret_tuple[1], ret_tuple[2], outmask
```

Comment:
You can just append outmask to the returned tuple:
```
if qdone is False:
        ret_tuple = fitfunc(ydata, outmask, arg_dict, **kwargs_optimizer)
return ret_tuple + (outmask,)
```

Reply:
Thanks, changed!